File: backend/app/schemas/integrations.py
"""Integration catalog schemas."""

from pydantic import BaseModel


class IntegrationSummary(BaseModel):
    key: str
    name: str
    category: str
    description: str
    status: str
    setup_hint: str
    capabilities: list[str]


class IntegrationCatalog(BaseModel):
    integrations: list[IntegrationSummary]

File: frontend/src/api/integrations.ts
import { get } from "./client";

export interface IntegrationSummary {
  key: string;
  name: string;
  category: string;
  description: string;
  status: "configured" | "available";
  setup_hint: string;
  capabilities: string[];
}

export function getIntegrations(): Promise<{ integrations: IntegrationSummary[] }> {
  return get<{ integrations: IntegrationSummary[] }>("/integrations");
}
